components/global_header/global_header.tsx

```diff
@@ -22,7 +22,7 @@ const GlobalHeaderContainer = styled.header`
     border-bottom: solid 1px rgba(var(--center-channel-color-rgb), 0.08);
     color: rgba(var(--global-header-text-rgb), 0.64);
     padding: 0 12px;
-    z-index: 100;
+    z-index: 3;
```

Comment:
this is creating issue with Product menu, have fixed it mine as Temporary solution.
I think we should increase the z-index of overlay and Task list

Reply:
+1

Reply:
I can optimise that, however we are causing fundamental issues by assigning a z-index to the global header. Potentially we shouldn't have to pass a z-index at all to inline elements especially if they aren't covered by other elements. We should be adjusting the z-indexing of absolute positioned elements only, pop ups, menus, overlays. I see a lot of issues when I was trying to implement menus, be it using popper or not, when global header has such a high z-index. 

As for what is covering the menu in Products when rendered in boards, does that banner need to have a higher z-index too? I did a small test removing it and it didn't change the UI. Can we try to remove redundant z-indexes so we can prevent issues with overlap in the future?